src/traces/parcats/calc.js

```diff
@@ -480,19 +480,19 @@ function validateDimensionDisplayInds(trace) {
 function validateCategoryProperties(dim, uniqueInfoDim) {
 
     // Update categoryarray
-    dim.categoryarray = uniqueInfoDim.uniqueValues;
+    dim._categoryarray = uniqueInfoDim.uniqueValues;
```

Comment:
looks reasonable, seems like we always go through this step so simply adding new `_` attributes should be safe (in fact this may fix a bug if there are any `plot` pathways that will redraw a `parcats` without hitting `calc`.

@jonmmease I notice the docstring for this function is copypasta... care to enlighten us as to what's really going on here?

Reply:
Yeah, sorry, this function has an unhelpful docstring and its name didn't evolve with the change in how category order is specified.  The purpose of this function is to set `categoryarray` and `ticktext` with values that are consistent with the unique values found for this dimension.

A better docstring would be something like
```
/**
 * Update category properties based on the unique values found for this dimension
 * ...
 */
```

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/3072/commits/3bb0a774623d32bd48c6db854f4e3d8b7ffcf0cd